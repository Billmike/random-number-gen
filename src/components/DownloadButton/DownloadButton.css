.download-button {
  height: 35px;
  border-radius: 10px;
  border: none;
  color: white;
  font-size: 16px;
  margin-left: 20px;
  font-family: auto;
}