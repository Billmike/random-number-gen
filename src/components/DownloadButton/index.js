import React from 'react';
import './DownloadButton.css';

const DownloadButton = (props) => {
  return (
    <button
      disabled={props.numbers.length === 0}
      className="download-button"
      onClick={props.onDownloadPhoneNumbers}
      style={{
        cursor: props.numbers.length === 0 ? 'not-allowed' : 'pointer',
        backgroundColor: props.numbers.length === 0 ? 'grey' : 'midnightblue'
      }}
    >
      Download Numbers
    </button>
  )
}

export default DownloadButton;
