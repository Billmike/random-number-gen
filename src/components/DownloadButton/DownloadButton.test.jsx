import React from 'react';
import { shallow } from 'enzyme';
import DownloadButton from './index';

test('should ', () => {
  const props = {
    numbers: [],
    onDownloadPhoneNumbers: jest.fn()
  }
  const wrapper = shallow(<DownloadButton {...props} />);

  expect(wrapper).toMatchSnapshot()
})
