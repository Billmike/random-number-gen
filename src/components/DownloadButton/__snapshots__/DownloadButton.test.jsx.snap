// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should  1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DownloadButton
    numbers={Array []}
    onDownloadPhoneNumbers={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Download Numbers",
      "className": "download-button",
      "disabled": true,
      "onClick": [MockFunction],
      "style": Object {
        "backgroundColor": "grey",
        "cursor": "not-allowed",
      },
    },
    "ref": null,
    "rendered": "Download Numbers",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Download Numbers",
        "className": "download-button",
        "disabled": true,
        "onClick": [MockFunction],
        "style": Object {
          "backgroundColor": "grey",
          "cursor": "not-allowed",
        },
      },
      "ref": null,
      "rendered": "Download Numbers",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
